### PR TITLE
Sk multi assemble

### PR DESCRIPTION
**Pull request by rewlad**
## Status
**READY/IN DEVELOPMENT/HOLD**

## Cross module movements
YES | NO

## Description
A few sentences describing the overall goals of the pull request's commits.

## Changes
- Feature 1
- Feature 2

## Impacted Areas in Application
List general components of the application that this PR will affect:

* Feature 1
* Feature 2

### CODE REVIEW

File: base_lib/src/main/scala/ee/cone/c4actor/HashSearchImpl.scala [hunk 1]
```diff
@@ -90,7 +90,8 @@ object HashSearchImpl {
       val(valueToRanges,byToRanges) = ranger.ranges(by)
       IndexerImpl(modelConditionFactory.filterMetaList(lens),by,this)(preHashing,modelClass,modelConditionFactory,lens.of,valueToRanges,byToRanges.lift)
     }
-    def assemble = new BHashSearchAssemble(modelClass,this, preHashing)
+    def assemble: BHashSearchAssemble[Model] =
+      new BHashSearchAssembleFactory(preHashing).create(modelClass,this)
```
Comment (IlyaHalsky) [inline base_lib/src/main/scala/ee/cone/c4actor/HashSearchImpl.scala]: Eh?

File: base_lib/src/main/scala/ee/cone/c4actor/dep_impl/DepReqByPKImpl.scala [hunk 2]
```diff
@@ -53,17 +54,23 @@ object ByPKTypes {
     askByPKs.distinctBy(_.forClassName).collect { case bc: AskByPKImpl[_] => bc.assemble }
 }
 
-@c4("ByPKRequestHandlerCompApp") case class AskByPKFactoryImpl(depAskFactory: DepAskFactory, util: DepResponseFactory) extends AskByPKFactory {
-  def forClass[A <: Product](cl: Class[A]): AskByPK[A] =
-    AskByPKImpl(cl.getName, util)(cl, depAskFactory.forClasses(classOf[N_ByPKRequest], classOf[List[A]]))
+@c4("ByPKRequestHandlerCompApp") case class AskByPKFactoryImpl(
+  depAskFactory: DepAskFactory,
+  byPKGenericAssembleFactory: ByPKGenericAssembleFactory
```
Comment (IlyaHalsky) [inline base_lib/src/main/scala/ee/cone/c4actor/dep_impl/DepReqByPKImpl.scala]: why factory is passed and not used

File: base_lib/src/main/scala/ee/cone/c4actor/HashSearchImpl.scala (hunk 1 above)
Comment (rewlad) [reply base_lib/src/main/scala/ee/cone/c4actor/HashSearchImpl.scala]: оно nested, hashsearch я решил на этот раз отложить

File: base_lib/src/main/scala/ee/cone/c4actor/dep_impl/DepReqByPKImpl.scala (hunk 2 above)
Comment (rewlad) [reply base_lib/src/main/scala/ee/cone/c4actor/dep_impl/DepReqByPKImpl.scala]: AskByPKImpl is 'case class', but has 'Class[A]' in second args list;
so it's not c4multi, and factory is manual

**Pull request merged**